### ISSUE TITLE
Angular2  @angular module imports packages with undesirable names

### ISSUE THREAD

**Issue by eastlondoner** [jspm/jspm-cli#1872]
I'm using the latest angular2 release candidate with jspm and it doesn't seem to play nicely:

e.g.
jspm install npm:@angular/router
results in
SystemJS.config({
map: {
"router": "npm:@angular/router@2.0.0-rc.1",
}

which means import * from @angular/router doesn't work and I have to manually add the @angular/router mapping to SystemJS

just using 'router' / 'core' / 'common' / 'http' / etc. instead of '@angular/router' ... '@angular/http'
is a pain because it's not "canonical" angular2 and some of the module names are things like 'http' which obviously causes a name collision issue.

as an aside: this is the @angular/router package.json
{
"name": "@angular/router",
....
so I am not sure where it is getting just "router" from - this seems to occur with all the @angular modules

**Comment by guybedford** [comment]
Try running `jspm install @angular/router=npm:@angular/router` here then rather.

**Comment by eastlondoner** [comment]
ahah, those are the magic runes I am looking for. Thanks!

**Comment by kylecordes** [comment]
@guybedford NPM scoped packages appear to be proliferating rapidly nowadays. How about an update to JSPM which makes them "just work" without the incantation above?

**Comment by guybedford** [comment]
@kylecordes this does work as expected for dependencies of npm packages, and we did actually used to do this, but the thing is that from the perspective of jspm, `jspm install github:components/jquery` and `jspm install npm:@angular/router` are the same type of case.

Previously we would install jquery from github into `components/jquery` so you would need to `import $ from 'components/jquery'`. This was changed to make it easier for users.

Typically also, with npm scoped packages the `@scope` is the organisation name. I personally think it's more succinct to use the default shortname to omit the organisation.

**Comment by kylecordes** [comment]
@guybedford Regarding the install command, is this driven by the desire to make the different install sources equal, to avoid preferring NPM as a default?

Regarding the scopes, unfortunately from what I can see in various tools adopting scoped packages, most of the example code out there assumes the full name including the scope. In other words it assumes the scope as part of the name, not the name of the containing organization. For example "@angular/core" is used, I don't think anyone wants to refer to this in code as "core". I use a tool called "@ngrx/store", and everyone calls it that, both in code and in text. I have seen no references as just "store".

(Unfortunately I don't have any communitywide data about such naming. It is currently rather hard to obtain it, because the NPMJS folks still think of scoped packages is a close relative of private packages, and therefore still exclude them from the search.)

Would a bit of special case be too terrible? Could it recognize this:

```
jspm install SRC:@SCOPE/PKG
```

and interpret it as something like:

```
jspm install @SCOPE/PKG=SRC:@SCOPE/PKG
```

?

I would describe the current behavior as being optimized for the past (things like jQuery) rather than optimized for the future (new stuff coming out as scoped NPM packages). That seems like an unfortunate optimization, my sense is that people using tools from the past are pretty happy doing so without any help from JSPM, and people using tools of the future direly need such help.

**Comment by aluanhaddad** [comment]
> NPMJS folks still think of scoped packages is a close relative of private packages, and therefore still exclude them from the search.

Doesn't that imply that the `@angular/package` scheme is a break from npm convention since it is published for external, that is to public, consumption?

**Comment by kylecordes** [comment]
The "@" marker originally meant "private". Now it means "scoped". We have seen the beginning of a trend that I would guess is going to expand dramatically, lots of public published scoped packages. Here is some scoped package info:

https://blog.nodejitsu.com/a-summary-of-scoped-modules-in-npm/

From this and other pages about the topic, it seems pretty clear that scoped no longer implies private, and that there is a general expectation of the overall community moving mostly toward scoped in the future.

I don't think tooling that chooses to treat "@" as private or as second-class is going to hold back the tide, it's just going to frustrate the people using that tooling. Or similarly: we live in a competitive market space. Scoped packages already "just work" in NPM proper, and thus "just work" for the webpack crowd.

**Comment by aluanhaddad** [comment]
@kylecordes thank you very much for the clarification. 

It does seem like this should be supported.

Personally, I was somewhat frustrated when the angular folks introduced this change because it seems to violate the conceptual principle of modularity that the framework claims to adhere to. I thought to myself well they may not have @Script but they got that silly-looking glyph in there somehow.

Perhaps this could be implemented as a general mechanism for some sort of nestable package configurations. There are other use cases for that which I think it could be helpful, such as applying granular "meta" configuration.

**Comment by guybedford** [comment]
These names have been cleared up in the latest jspm 0.17 beta.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.18.
